components/Sidebar.tsx

```diff
@@ -76,7 +78,7 @@ const Sidebar: React.FC<Props> = ({ menuState, nav: currentNav }) => {
             router.push(s.link, undefined, { shallow: false });
         }
     };
-
+useEffect(() => setTech(menuItem[indexOf]), [indexOf])
```

Comment:
should use callback => useEffect(()=>useCallback(setTech

Reply:
why usecallback inside useEffect 🤔

Reply:
for memoized function return?

Reply:
Do we need to memoize setState calls?